src/jit/morph.cpp

```diff
@@ -1689,6 +1701,40 @@ void fgArgInfo::ArgsComplete()
     argsComplete = true;
 }
 
+#if defined(UNIX_X86_ABI)
+void fgArgInfo::ArgsAlignPadding()
+{
+    // To get the padding amount, sum up all the slots and get the remainer for padding
+    unsigned         curInx;
+    unsigned         numSlots         = 0;
+    fgArgTabEntryPtr firstArgTabEntry = nullptr;
+
+    for (curInx = 0; curInx < argCount; curInx++)
+    {
+        fgArgTabEntryPtr curArgTabEntry = argTable[curInx];
+        if (curArgTabEntry->numSlots > 0)
+        {
+            // the argument may be REG_STK or constant or register that goes to stack
+            assert(nextSlotNum >= curArgTabEntry->slotNum);
+
+            numSlots += curArgTabEntry->numSlots;
+            if (firstArgTabEntry == nullptr)
+            {
+                // first argument will be used as for add padding
+                firstArgTabEntry = curArgTabEntry;
+            }
+        }
+    }
+    // calc padding to align
+    const int numSlotsAligned = STACK_ALIGN / TARGET_POINTER_SIZE;
+    if (firstArgTabEntry != nullptr)
+    {
+        firstArgTabEntry->padStkAlign = (numSlotsAligned - (numSlots % numSlotsAligned)) % numSlotsAligned;
+        this->padStkAlign             = firstArgTabEntry->padStkAlign;
```

Comment:
When numSlots is 8, it would be 4 - (8 % 4) = 4. If this happens, we don't need to add as it is already aligned. So we need the last `% numSlotsAligned`